yandex_contest/bin_sum.py

```diff
@@ -0,0 +1,53 @@
+def convert(string):
+    list_1 = []
+    list_1[:0] = string
+    list_x = [int(i) for i in list_1]
+    return list_x
+
+def main():
+    num_1 = input()
+    list_1 = convert(num_1)
+
+    num_2 = input()
+    list_2 = convert(num_2)
+
+    result = []
+    summa = 0
+    rest = 0
+
+    number1 = list(reversed(list_1))
+    number2 = list(reversed(list_2))
+
+    diff = len(list_1) - len(list_2)
+    if diff < 0:
+        for _ in range(abs(diff)):
+            number1.append(0)
+    elif diff > 0:
+        for _ in range(abs(diff)):
+            number2.append(0)
```

Comment:
There is a nice trick to simplify code like this one.
The main problem here is that you don't know what number is longer. But if you know that `list_1` is always shorter (or of the same length) than `list_2` this code be simplified to just
```python3
for _ in range(len(list_2) - len(list_1)):
    number1.append(0)
```
Try to think about how to guarantee that `list_1` is always shorter than `list_2`?